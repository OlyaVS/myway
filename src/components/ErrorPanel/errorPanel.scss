@import '../../style/themes.scss';

.error-panel {
  position: relative;
  width: 100%;
  height: 30px;
  padding:0 5px;
  margin-bottom: 20px;
  font-size: 14px;
  color: map-get($error-panel, text);
  text-align: center;
  border-radius: 5px;
  background-color: map-get($error-panel, bg);
  overflow: hidden;
}
