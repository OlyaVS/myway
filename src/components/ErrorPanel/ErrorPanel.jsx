import React from 'react';
import PropTypes from 'prop-types';

import './errorPanel.scss';

function ErrorPanel({ error }) {
  return error ? (
    <div className={'error-panel'} data-testid={'error-panel'}>
      <span className={'error-panel__text'}>{error}</span>
    </div>
  ) : null;
}

ErrorPanel.propTypes = {
  error: PropTypes.any.isRequired,
};

export default ErrorPanel;
