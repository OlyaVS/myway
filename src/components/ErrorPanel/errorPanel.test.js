import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ErrorPanel from './ErrorPanel';

describe('Error panel', () => {
  afterEach(cleanup);

  test('Does not exist when no error', () => {
    const { container } = render(<ErrorPanel error={false} className={'error-panel'} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Renders with error message when error occurs', () => {
    const routeError = 'Address was not added. Please try agan.';
    const { getByTestId } = render(<ErrorPanel error={routeError} className={'error-panel'} />);
    const errorPanel = getByTestId('error-panel');
    expect(errorPanel.textContent).toEqual(routeError);
  });
});
