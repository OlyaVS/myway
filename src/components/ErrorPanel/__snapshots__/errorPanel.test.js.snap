// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error panel Does not exist when no error 1`] = `null`;
